Samples/MobileBuyIntegration/MobileBuyIntegration/Views/ProductView.swift

```diff
@@ -47,39 +47,42 @@ struct ProductView: View {
-                        Rectangle()
-                            .fill(Color.gray.opacity(0.2))
-
-                        AsyncImage(url: imageURL) { phase in
-                            switch phase {
-                            case .empty:
-                                EmptyView()
-                            case let .success(image):
-                                image
-                                    .resizable()
-                                    .aspectRatio(contentMode: .fill)
-                                    .frame(height: 400)
-                                    .clipped()
-                                    .opacity(imageLoaded ? 1 : 0)
-                                    .onAppear {
-                                        withAnimation(.easeIn(duration: 0.5)) {
-                                            imageLoaded = true
-                                        }
+                    AsyncImage(url: imageURL.thumbnail(size: 600)) { phase in
+                        switch phase {
+                        case .empty:
+                            Rectangle()
+                                .fill(Color.gray.opacity(0.2))
+                                .frame(height: 400)
+                        case let .success(image):
+                            image
+                                .resizable()
+                                .aspectRatio(contentMode: .fill)
+                                .frame(height: 400)
+                                .clipped()
+                                .opacity(imageLoaded ? 1 : 0)
+                                .onAppear {
+                                    withAnimation(.easeIn(duration: 0.5)) {
+                                        imageLoaded = true
                                     }
-                            case .failure:
-                                Image(systemName: "photo")
-                                    .resizable()
-                                    .frame(width: 100, height: 100)
-                                    .foregroundColor(.gray)
-                            @unknown default:
-                                EmptyView()
-                            }
+                                }
+                        case .failure:
+                            Rectangle()
+                                .fill(Color.gray.opacity(0.2))
+                                .frame(height: 400)
+                                .overlay(
+                                    Image(systemName: "photo")
+                                        .resizable()
+                                        .frame(width: 100, height: 100)
+                                        .foregroundColor(.gray)
+                                )
+                        @unknown default:
+                            Rectangle()
+                                .fill(Color.gray.opacity(0.2))
+                                .frame(height: 400)
```

Comment:
I'm fixing the layout shift when the product images load in the `ProductView` here